scripts/appscript.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+set -eu -o pipefail
+
+[[ $# -lt 1 ]] && {
+    echo "Usage $0 <BUILD_SLUG>" >&2
+    echo "  Example: $0 bucket-foo/randomid" >&2
+    exit 1
+}
+
+# Required vars
+BUILD_SLUG=$1
+VAULT_VERSION=$2
+DYNAMODB_TABLE=$3
+KSM_KEY_ID=$4
+SSM_PATH=$5
+
+# Internal vars
+AWS_AZ=$(curl -sSL http://169.254.169.254/latest/meta-data/placement/availability-zone)
+AWS_DOMAIN=$(curl -sSL http://169.254.169.254/latest/meta-data/services/domain)
+SALT_SLUG="${BUILD_SLUG}/salt"
+
+# Export standard aws envs
+export AWS_DEFAULT_REGION=${AWS_AZ:0:${#AWS_AZ} - 1}
+
+# Export Vault local address
+export VAULT_ADDR=http://127.0.0.1:8200
+
+#Update salt-call to
+sudo yum update salt-master
+
+echo "[appscript]: Ensuring default salt srv location exists, /srv/salt..."
+mkdir -p /srv/salt
+
+echo "[appscript]: Syncing salt content from s3://${SALT_SLUG}..."
+aws s3 sync --delete "s3://${SALT_SLUG}" /srv/salt
+
+echo "[appscript]: Updating salt grains..."
+salt-call --local saltutil.sync_grains
+
+echo "[appscript]: Configuring salt to read ec2 metadata into grains..."
+echo "metadata_server_grains: True" > /etc/salt/minion.d/metadata.conf
+
+echo "[appscript]: Setting required salt grains for vault..."
+salt-call --local grains.setval vault \
+"{'version':'${VAULT_VERSION}', 'dynamodb_table':'${DYNAMODB_TABLE}', 'kms_key_id':'${KSM_KEY_ID}', 'region':'${AWS_DEFAULT_REGION}', 'ssm_path': '${SSM_PATH}'}"
+
+echo "[appscript]: Applying the vault install and configure states..."
+salt-call --local --retcode-passthrough state.sls vault -l debug 2>&1 | tee /var/log/salt_vault_log
```

Comment:
recommend using a file extension
```suggestion
salt-call --local --retcode-passthrough state.sls vault -l debug 2>&1 | tee /var/log/salt_vault.log
```